packages/react/src/components/TableHandles/ExtendButton/ExtendButton.tsx

```diff
@@ -1,14 +1,41 @@
 import {
+  BlockFromConfigNoChildren,
+  DefaultBlockSchema,
   DefaultInlineContentSchema,
   DefaultStyleSchema,
+  EMPTY_CELL_HEIGHT,
+  EMPTY_CELL_WIDTH,
   InlineContentSchema,
+  mergeCSSClasses,
   PartialTableContent,
   StyleSchema,
 } from "@blocknote/core";
-import { MouseEvent as ReactMouseEvent, useEffect, useState } from "react";
+import {
+  MouseEvent as ReactMouseEvent,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useState,
+} from "react";
+import { RiAddFill } from "react-icons/ri";
 
 import { useComponentsContext } from "../../../editor/ComponentsContext.js";
-import { TableHandleProps } from "../TableHandleProps.js";
+import { ExtendButtonProps } from "./ExtendButtonProps.js";
+
+// Rounds a number up or down, depending on whether the value past the decimal
+// point is above or below a certain fraction. If no fraction is provided, it
+// behaves like Math.round.
+const roundUpAt = (num: number, fraction = 0.5) => {
+  if (fraction <= 0 || fraction >= 100) {
```

Comment:
100 seems wrong. Also, the current implementation is more like a threshold. passing 2.3 for example will always return 3?